Remove superfluous realpath call

tkn98 opened the pull request:
Calling realpath is not necessary for __DIR__

Refs:

- http://php.net/manual/language.constants.predefined.php

(Previously reported against wrong base branch in #5518)

tkn98 committed:
Remove superfluous realpath call

Calling realpath is not necessary for __DIR__

Refs:

- http://php.net/manual/language.constants.predefined.php

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=9025) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=9025) before we can accept your contribution.<br/><hr/>**Tom Klingenberg** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

okorshenko commented on pub/index.php:
```diff
@@ -10,7 +10,7 @@
 use Magento\Framework\App\Filesystem\DirectoryList;
 
 try {
-    require realpath(__DIR__) . '/../app/bootstrap.php';
```
It make sense to change the code in the following way:
`realpath(__DIR__ . '/../app/bootstrap.php')`
in this case we will get `/root/folder/app/bootstrap.php` insead of `/root/folder/pub/../app/bootstrap.php`

tkn98 replied:
On closer look I'd say that such a suggestion is not necessary. Require deals fine with such paths, it does not need any further assistance. Or do you have a scenario at hand where that is not the case? I personally don't know of any nor can I imagine one.

orlangur replied:
Passing whole string to `realpath` is a bad idea from debugging perspective, see http://stackoverflow.com/a/21035675.

I saw a lot `realpath` calls all over the code, but for `__FILE__` or `__DIR__` it seems useless.

okorshenko commented:
@tkn98 could you please sign CLA? Thank you

tkn98 commented:
@okobchenko: reg. CLA: Wrong author-email was in the commit, that's why the CLA was shown not-signed. I have signed that long ago, should come green then, updated the commits author.

tkn98 commented:
@okobchenko: reg. CLA: (update: now fixed) Wrong author-email was in the commit, that's why the CLA was shown not-signed. I have signed that long ago, should come green then, updated the commits author.

okorshenko commented:
@tkn98 thank you for your contribution. Your changes successfully merged https://github.com/magento/magento2/commit/485adc5b9c245cd67a99255b2d2da324263deb50#diff-4aa58a2cf6c6b39adfc4e63689aea3c0
